### DISCUSSION TITLE
Marketplace ToU violates GPL and similar licenses?

### DISCUSSION THREAD

**Discussion by xsrvmy** [microsoft/vsmarketplace#322] [General]
The marketplace hosts some GPL extensions, but at the same time the ToU restricts the use of downloaded extensions to official products only, and it does not contain the usual disclaimer that this restriction does not apply to open source code. Doesn't this violate the GPL license? (Also it makes it dubious as to whether extensions can contain third party GPL code, which is why I am asking this in the first place.)

**Comment by xsrvmy** [comment]
Example of a GPL extension: kde.breeze, AdaCore.ada
Example of an LGPL extension: uloco.bluloco-theme-light
There are some example that require third party GPL code as well.

**Comment by prashantvc** [comment]
VS Marketplace TOUs govern how a user is permitted to use the marketplace, not how a user is permitted to use extensions.  Publishers agree (per a separate publisher agreement) only to upload extensions that extend the capabilities of Microsoft Visual Studio, Visual Studio for Mac, Visual Studio Code, GitHub Codespaces, Azure DevOps, Azure DevOps Server, and successor products and services; but the license a publisher associates with its extension governs how an end user is permitted to use that extension, not the Marketplace TOU or the publisher agreement.

Hope it answers your concern

**Comment by xsrvmy** [comment]
The problematic sentence is this: "Marketplace 
Offerings are intended for use only with Visual Studio Products and Services and you may only install 
and use Marketplace Offerings with Visual Studio Products and Services."

**Reply by ssbarnea** [reply]
While I do agree that this sentence is problematic, towards preventing the freedom of choice, it is not licensing, it is part of ToS (term of service).

Let's not hide behind the bush, we probably know that they are afraid of a fork that could potentially be more open and not controlled by them that would piggy-back marketplace and install extensions from it. That is likely why https://open-vsx.org/ was created as an open alternative to a locked-in system. Some publishers publish to both and I doubt we will ever see any rule preventing publishing in another place,... as the image damage would be huge.

PS. I am not a lawyer.
